### DISCUSSION TITLE
Question about spmv

### DISCUSSION THREAD

**Discussion by zhujj2008** [NVIDIA/cccl#1068] [CUB]
Hi, 
      I read the source code of spmv::CsrMV and found that many optimizations have been made. I am very curious whether the implementation in the cuSparse library also uses the same optimization method? Thanks。

**Comment by malmasri7** [comment]
Hi @zhujj2008
Overall, CUSPARSE uses a better load-balancing strategy and optimizations; thus, it should be faster than the CUB implementation. Moreover, cusparseSpMV will have a newer and improved CSR SpMV implementation in CUDA 12.1.

**Reply by zhujj2008** [reply]
Thanks @malmasri7 .
Is there any paper or blog published by Nvidia about the optimization strategy of cusparse csr spmv？
